src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -640,7 +663,34 @@ private Article entityToArticle(Entity entity){
     return article;
   }
 
-  public void updateFieldForum(String forumName, String fieldToAppend, String valueToAppend, boolean updateAll){
+  public void updateFieldForum(String forumName, String fieldToRemove, String valueToRemove, boolean updateAll){
+    PreparedQuery results = null;
+    if(updateAll){
+      Query query =
+              new Query("Forum")
+              .addSort("title",SortDirection.ASCENDING);
+      results = datastore.prepare(query);
+    }
+    else{
+      Query query =
+              new Query("Forum")
+                      .setFilter(new Query.FilterPredicate("title", FilterOperator.EQUAL, forumName));
+      results = datastore.prepare(query);
+    }
+      int count = 0;
+
+    for(Entity forumEntity : results.asIterable()) {
+        count++;
+      String fieldBeingUpdated = (String) forumEntity.getProperty(fieldToRemove);
+      fieldBeingUpdated+=","+valueToRemove;
+      forumEntity.setProperty(fieldToRemove,fieldBeingUpdated);
```

Comment:
are you removing a field? or updating one?

Reply:
Why are the parameters called "fieldToRemove" etc?

Reply:
My bad on this one. I was duplicating and editing updateFieldForum to removeFieldForum. Looks like I accidentally replaced all the names.